backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -444,15 +441,73 @@ public Response createOrUpdate(@PathParam("vaultId") UUID vaultId, @Valid @NotNu
 		}
 	}
 
+	@POST
+	@Path("/{vaultId}/claim-ownership")
+	@RolesAllowed("user")
+	@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+	@Transactional
+	@Operation(summary = "claims ownership of a vault",
+			description = "Assigns the OWNER role to the currently logged in user, who proofs this claim by sending a JWT signed with a private key held by users knowing the Vault Admin Password")
+	@APIResponse(responseCode = "200", description = "ownership claimed successfully")
+	@APIResponse(responseCode = "400", description = "incorrect proof")
+	@APIResponse(responseCode = "404", description = "no such vault")
+	@APIResponse(responseCode = "409", description = "owned by another user")
+	public Response claimOwnership(@PathParam("vaultId") UUID vaultId, @FormParam("proof") @Valid @ValidJWS String proof) {
+		User currentUser = User.findById(jwt.getSubject());
+		Vault vault = Vault.<Vault>findByIdOptional(vaultId).orElseThrow(NotFoundException::new);
+
+		// if vault.authenticationPublicKey no longer exists, this vault has already been claimed by a different user
+		var authPubKey = vault.getAuthenticationPublicKey().orElseThrow(() -> new ClientErrorException(Response.Status.CONFLICT));
+
+		try {
+			var verifier = JWT.require(Algorithm.ECDSA384(authPubKey))
+					.acceptLeeway(30)
+					.withClaimPresence("nbf")
+					.withClaimPresence("exp")
+					.withSubject(currentUser.id)
+					.withClaim("vaultId", vaultId.toString().toLowerCase())
+					.build();
+			verifier.verify(proof);
+		} catch (JWTVerificationException e) {
+			throw new BadRequestException("Invalid proof of ownership", e);
+		}
+
+		Optional<VaultAccess> existingAccess = VaultAccess.findByIdOptional(new VaultAccess.Id(vaultId, currentUser.id));
+		if (existingAccess.isPresent()) {
+			var access = existingAccess.get();
+			access.role = VaultAccess.Role.OWNER;
+			access.persist();
+			AuditEventVaultMemberUpdate.log(currentUser.id, vaultId, currentUser.id, VaultAccess.Role.OWNER);
+		} else {
+			var access = new VaultAccess();
+			access.vault = vault;
+			access.authority = currentUser;
+			access.role = VaultAccess.Role.OWNER;
+			access.persist();
+			AuditEventVaultMemberAdd.log(currentUser.id, vaultId, currentUser.id, VaultAccess.Role.OWNER);
+		}
+
+		vault.salt = null;
+		vault.iterations = null;
+		vault.masterkey = null;
+		vault.authenticationPrivateKey = null;
+		vault.authenticationPublicKey = null;
+		vault.persist();
+
+		// TODO: log some event?
```

Comment:
I think it would make sense to introduce a new audit event. "Update Vault" doesn't really reflect this important change. How should we call this? "Deprecate Vault Admin Password"?

Reply:
Claimed ownership?

Reply:
I already introduced an "Update Vault Member" audit event that shows which user claimed ownership. This log event is for "nulling" the vault admin password fields, not really for claiming ownership. That's why "Update Vault" would fit better, but... it wouldn't then tell anyone anything about the "removal" of the vault admin password.

Reply:
I'm leaning towards "Remove Vault Admin Password".

Reply:
That is a rather technical perspective. From the user's point of view he never removed a password, he pressed "claim ownership".
Thus I wouldn't split this into two events.

Reply:
Makes sense. It may confuse admins when the audit event doesn't match the action from a user perspective. Then I'll introduce a new "Claim Vault Ownership" audit event.

Regarding "two" events, I was just filling in the TODOs, to be honest. Shall I delete `AuditEventVaultMemberUpdate.log()`/`AuditEventVaultMemberAdd.log()` in the lines before that?

Reply:
Not sure, we have these events during vault creation as well. So maybe "two" is not the problem, as long as the events can be related via time/actor.

Reply:
Fixed in 92b4fb35ebabc48f61baa705122e78cc23a37259.